mail/api.py

```diff
@@ -279,18 +279,31 @@ def build_message(connection, template_name, recipient, context, metadata=None):
     return msg
 
 
-def send_messages(messages):
+def send_messages(messages, raise_errors=False):
     """
     Sends the messages and logs any exceptions
 
     Args:
         messages (list of django.core.mail.EmailMultiAlternatives): list of messages to send
+        raise_errors (bool): flag specifying if this method should raise errors
+
+    Raises:
+        EmailSendFailureException: Raised if any of the email send is failed
     """
+    errored_emails = set()
+
     for msg in messages:
         try:
             msg.send()
         except:  # pylint: disable=bare-except
-            log.exception("Error sending email '%s' to %s", msg.subject, msg.to)
+            errored_emails.add(msg)
+    if errored_emails:
+        for email_msg in errored_emails:
+            log.exception(
```

Comment:
`log.exception()` shouldn't be called outside of a `try ... except ...` block
```suggestion
            log.error(
```